src/generic/MPoly.jl

```diff
@@ -497,6 +497,18 @@ function Base.hash(x::MPoly{T}, h::UInt) where {T <: RingElement}
    return b
 end
 
+# Fallback hash function for multivariate polynomials implemeting the
+# iterators
+function Base.hash(x::MPolyElem{T}, h::UInt) where {T <: RingElement}
+   b = 0x53dd43cd511044d1%UInt
+   for (e, c) in zip(exponent_vectors(x), coeffs(x))
+      b = xor(b, xor(hash(c, h), h))
+      b = xor(b, xor(hash(e, h), h))
+      b = (b << 1) | (b >> (sizeof(Int)*8 - 1))
```

Comment:
Why this operation? I realize a similar line is in the implemation of `hash` in Poly.jl, so is it for some sort of compatibility (i.e. to respect `isequal` between polys of different types)?

Reply:
I don't know. I just copy it whenever I write a hash function :)

It is definitely not for compability.

Reply:
Ok, so it seems useless but also harmless :)